Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,44 +1,80 @@
-import React from 'react';
+import React, { Component } from 'react';
 import CalculatorDisplay from './components/DisplayComponents/CalculatorDisplay';
 import ActionButton from './components/ButtonComponents/ActionButton';
 import NumberButton from './components/ButtonComponents/NumberButton';
 import './App.css';
 
-const App = () => {
-  return (
-    <div className='main__container'>
-      <CalculatorDisplay />
-      <div className='button__container'>
-        <div className='action__container'>
-          <ActionButton text='clear' />
-          <NumberButton digit='7' />
-          <NumberButton digit='8' />
-          <NumberButton digit='9' />
-          <NumberButton digit='4' />
-          <NumberButton digit='5' />
-          <NumberButton digit='6' />
-          <NumberButton digit='1' />
-          <NumberButton digit='2' />
-          <NumberButton digit='3' />
-          <ActionButton text='0' />
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.initialState = { display: '' }
+    this.state = this.initialState;
+  }
+
+  displayHandler = value => {
+if(value === 'X') {
+  value = '*';
+}
+    this.setState({
+      display: this.state.display + value
+    })
+  }
+
+  clearDisplay = () =>{
+    this.setState({
+      display: ''
+    })
+  }
+
+  evaluate = () =>{
+    console.log(this.state.display.toString());
+    this.setState({
+      display: eval(this.state.display.toString())
+    })
```

Comment:
This is a nice simple solution to the functionality problem. 

I'm sure you know this already, but generally .eval shouldn't be used as it allows arbitrary execution of code, this introduces performance issues and security vulnerabilities - in-fact MDN say: "Do not ever use eval!".

However, it's still the most sensible option here I believe - I used it when I did it too :) Just something to be aware of.